### PR TITLE
[SYCLomatic] Adding sort_keys sort_pairs tests

### PR DESCRIPTION
**Pull request by danhoeflinger**
Adding tests for dpct::sort_pairs, dpct::sort_keys, and dpct::internal::translate_key.

Associated changes for these PR in SYCLomatic:
https://github.com/oneapi-src/SYCLomatic/pull/341 (merged)
https://github.com/oneapi-src/SYCLomatic/pull/468 (merged)
https://github.com/oneapi-src/SYCLomatic/pull/588 (merged)

Cannot pass until the above PRs are merged, as it is testing new functionality introduced there.

### CODE REVIEW

**Review by jzc — APPROVED** [review]

File: help_function/src/onedpl_test_sort_keys.cpp [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+    ASSERT_EQUAL(true, test_input2 == test_output1, "swapped input == output");
+  }
+
+  std::cout << std::endl
+            << test_suites_failed << " failing test(s) detected." << std::endl;
+  if (test_suites_failed == 0) {
+    return 0;
+  }
+  return 1;
+}
```
Comment (yihwang-nv) [inline help_function/src/onedpl_test_sort_keys.cpp]: Please add new line at the end of file.

File: help_function/src/onedpl_test_sort_pairs.cpp [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+    tests_failed +=
+        setup_and_run_pingpong<double, int,
+                               decltype(verify_double_int_descending),
+                               decltype(verify_double_int_ascending)>(
+            100, verify_double_int_descending, verify_double_int_ascending,
+            begin_bit, end_bit);
+
+    test_suites_failed += test_passed(tests_failed, test_name);
+  }
+}
```
Comment (yihwang-nv) [inline help_function/src/onedpl_test_sort_pairs.cpp]: Please add new line at the end of file.

File: help_function/src/onedpl_test_translate_key.cpp [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    test_suites_failed += test_passed(tests_failed, test_name);
+  }
+
+  std::cout << std::endl
+            << test_suites_failed << " failing test(s) detected." << std::endl;
+  if (test_suites_failed == 0) {
+    return 0;
+  }
+  return 1;
+}
```
Comment (yihwang-nv) [inline help_function/src/onedpl_test_translate_key.cpp]: Please add new line at the end of file.

**Review by yihwang-nv — APPROVED** [review]

File: help_function/src/onedpl_test_sort_keys.cpp (hunk 1 above)
Comment (danhoeflinger) [reply help_function/src/onedpl_test_sort_keys.cpp]: Added for all 3.

**Pull request merged**